models/woofi/avalanche_c/woofi_avalanche_c_trades.sql

```diff
@@ -0,0 +1,165 @@
+{{ config
+(
+    alias ='trades',
+    partition_by = ['block_date'],
+    materialized = 'incremental',
+    file_format = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['block_date', 'blockchain', 'project', 'version', 'tx_hash', 'evt_index', 'trace_address'],
+    post_hook='{{ expose_spells(\'["ethereum"]\',
+                                    "project",
+                                    "woofi",
+                                    \'["scoffie"]\') }}'
+)
+}}
+    
+
+{% set project_start_date = '2022-04-27' %}
+
+WITH dexs as 
+ (SELECT
+            evt_block_time AS block_time
+            ,'woofi' AS project
+            ,'1' AS version
+            ,from AS taker
+            ,to AS maker
+            ,fromAmount AS token_bought_amount_raw
+            ,toAmount AS token_sold_amount_raw
+            ,cast(NULL as double) AS amount_usd
+            ,fromToken AS token_bought_address
+            ,toToken AS token_sold_address
+            ,contract_address AS project_contract_address
+            ,evt_tx_hash AS tx_hash
+            ,'' AS trace_address
+            ,evt_index
+        FROM
+            {{ source('woofi_avalanche_c', 'WooPP_evt_WooSwap')}}
+        WHERE from <> '0x5aa6a4e96a9129562e2fc06660d07feddaaf7854' -- woorouter
+
+        {% if is_incremental() %}
+        AND evt_block_time >= date_trunc("day", now() - interval '1 week')
+        {% endif %}
+
+        UNION ALL 
+
+        SELECT
+            evt_block_time AS block_time
+            ,'woofi' AS project
+            ,'1' AS version
+            ,from AS taker
+            ,to AS maker
+            ,fromAmount AS token_bought_amount_raw
+            ,toAmount AS token_sold_amount_raw
+            ,cast(NULL as double) AS amount_usd
+            ,fromToken AS token_bought_address
+            ,toToken AS token_sold_address
+            ,contract_address AS project_contract_address
+            ,evt_tx_hash AS tx_hash
+            ,'' AS trace_address
+            ,evt_index
+        FROM
+            {{ source('woofi_avalanche_c', 'WooRouterV2_evt_WooRouterSwap')}}
+
+        {% if is_incremental() %}
+        WHERE evt_block_time >= date_trunc("day", now() - interval '1 week')
+        {% endif %}
+
```

Comment:
Hi, I think this part of the data WooRouterV2_evt_WooRouterSwap and WooPP_evt_WooSwap are duplicates. 
For example, the following example is obtained from WooRouterV2_evt_WooRouterSwap, but the same row of data exists in WooPP_evt_WooSwap.

https://snowtrace.io/tx/0x278d855ce63b3e3957762b5a7b10f28d81e26f6f5cec987b2be088905da131f5#eventlog

```sql
select * from woofi_avalanche_c.WooPP_evt_WooSwap where evt_tx_hash = "0x278d855ce63b3e3957762b5a7b10f28d81e26f6f5cec987b2be088905da131f5"
```

Reply:
![image](https://user-images.githubusercontent.com/10026361/200991439-82421b0f-6814-44ec-a564-f1357c750f20.png)

Reply:
So @LeviScoffie  try to filter out this part by code in line 37
```
FROM
            {{ source('woofi_avalanche_c', 'WooPP_evt_WooSwap')}}
        WHERE from <> '0x5aa6a4e96a9129562e2fc06660d07feddaaf7854' -- woorouter
```
Do you think this will work

Reply:
Oh, I don't quite understand why not only use WooPP_evt_WooSwap.

Reply:
Well the protocol uses  a maker that is not part of the Woo Swap protocol. when the market maker runs out of liquidiity, the trades are routed to a third party DEX. So i differentiated the tables so that we could get the trades that went direct to the main contract anad the ones that use the router.Although  nearly all the trades first go through the main contract @Hosuke @zhongyiio .

Reply:
> So @LeviScoffie try to filter out this part by code in line 37
> 
> ```
> FROM
>             {{ source('woofi_avalanche_c', 'WooPP_evt_WooSwap')}}
>         WHERE from <> '0x5aa6a4e96a9129562e2fc06660d07feddaaf7854' -- woorouter
> ```
> 
> Do you think this works for the case?

yes I filtetred using the router's address

Reply:
Understood, thanks for the answer.